src/ansys/dynamicreporting/core/utils/report_remote_server.py

```diff
@@ -1033,14 +1047,118 @@ def load_templates(self, templates_json):
                 break
 
         root_attr = templates_json[root_id_str]
-        root_template = self._populate_template(root_attr, None)
+        root_template = self._populate_template(root_id_str, root_attr, None)
         self.put_objects(root_template)
         id_template_map = {}
         id_template_map[root_id_str] = root_template
         self._build_templates_from_parent(root_id_str, id_template_map, templates_json)
         self._update_changes(root_id_str, id_template_map, templates_json)
 
 
+def _check_template(template_id_str, template_attr):
+    # Check template_id_str
+    if not _check_template_name_convention(template_id_str):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_id_str}' as the key.\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    # Check parent and children template name convention
+    if not _check_template_name_convention(template_attr["parent"]):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_attr['parent']}' "
+            f"that does not have the correct name convection under the key: 'parent' of '{template_id_str}'\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    for child_name in template_attr["children"]:
+        if not _check_template_name_convention(child_name):
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file has an invalid template name: '{child_name}' "
+                f"that does not have the correct name convection under the key: 'children' of '{template_id_str}'\n"
+                "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+            )
+
+    # Check key names
+    agreed_keys = {
+        "name",
+        "report_type",
+        "date",
+        "tags",
+        "params",
+        "sort_selection",
+        "item_filter",
+        "parent",
+        "children",
+    }
+    for key in template_attr.keys():
+        if key not in agreed_keys:
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file is using an unknown key: '{key}' other than any in the JSON schema.\n"
```

Comment:
@zhang-yuanrui can we make this into a warning and not an error that throws an exception? I'd like to keep the possibility open to use this format to transfer more information, if need be, in mechanisms outside of this pipeline. In this pipeline, if there are keys we don't recognize, we simply skip them.
Can we log the warning into the logger object that is defined in the Service class in adr_service?
Thanks.

Reply:
I can skip throwing an exception in report_remote_server level. But logging the warning in the Service class should be implemented in the next PR in the adr_service level. Do you think that's fair?